fdbclient/ThreadSafeTransaction.actor.cpp

```diff
@@ -368,4 +391,16 @@ ThreadFuture<Reference<ICluster>> ThreadSafeApi::createCluster(const char *clust
 	return ThreadSafeCluster::create(clusterFilePath, apiVersion);
 }
 
+void ThreadSafeApi::addNetworkThreadCompletionHook(void (*hook)(void*), void *hook_parameter) {
+	if (!g_network) {
+		throw network_not_setup();
+	}
+
+	auto hookPair = std::pair<void (*)(void*), void*>(hook, hook_parameter);
+
+	MutexHolder holder(lock); // We could use the network thread to protect this action, but then we can't guarantee upon return that the hook is set.
```

Comment:
I'm not sure I totally understand this comment.

Reply:
Unless it's just saying that this won't necessarily happen on the network thread.

Reply:
Running it on the network thread would allow us to skip the lock (and it's how we serialize most things in ThreadSafe), but it seemed more sensible to me that the hook be definitely set when this function returned. Running it on the network thread would instead have it be done at some point in the future.